### PR TITLE
Implemented Salesitems in Deal

### PR DESCRIPTION
**Pull request by mkeymolen**
It is now possible to retrieve the salesitems of a deal

### CODE REVIEW

File: src/TeamleaderDotNet/TeamleaderProductsApi.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        /// <summary>
+        /// Get all products from teamleader.
+        /// </summary>
+        /// <returns>A list of maximum 100 products from teamleader</returns>
+        public async Task<List<Product>> GetProducts()
+        {
+            return await DoCall<List<Product>>("getProducts.php", new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>("amount", "100"),
+                new KeyValuePair<string, string>("pageno", "0")
```
Comment (jeroenverh) [inline src/TeamleaderDotNet/TeamleaderProductsApi.cs:24]: Wouldn't it be better if we accept the page-nr via the method (if needed we can make it an optional field). Something like this:   GetProducts(int pageNr = 0, int amount = 100). 

We do need to validate the input (pagenr >= 0 and amount between 1 and 100), but then we can also support the use case where someone has more than 100 products in a single go.

**Comment by mkeymolen** [comment]
@jeroenverh do you have remarks or questions about the pull request? in the meantime, i've added some other improvements. I'm using the changes already in production via a lib reference, but i want to use the nuget package if possible..

**Comment by jeroenverh** [comment]
Yes,see my comment on src/TeamleaderDotNet/TeamleaderProductsApi.cs:

_Wouldn't it be better if we accept the page-nr via the method (if needed we can make it an optional field). Something like this: GetProducts(int pageNr = 0, int amount = 100).

We do need to validate the input (pagenr >= 0 and amount between 1 and 100), but then we can also support the use case where someone has more than 100 products in a single go._

**Comment by jeroenverh** [comment]
Oeps, just noticed that I forgot to click 'submit review'.. That's why you didn't see it.. Sorry :)

**Comment by jeroenverh** [comment]
@mkeymolen did you notice my feedback on your pull request?

**Comment by mkeymolen** [comment]
@jeroenverh yes, i'll take a look when i have some time, sorry. It is very busy right now dispite the holidays but i'll find some time next week i guess.

**Comment by jeroenverh** [comment]
Sure, no worries :) Whenever you have some time :)

**Pull request merged**